externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/schema/RowSet.scala

```diff
@@ -237,46 +238,32 @@ object RowSet {
         // Only match string in nested type values
         "\"" + s + "\""
 
-      // for Array Map and Row, temporarily convert to string
-      // TODO further analysis of type
-      case (list: java.util.List[_], _) =>
-        formatValue(list)
+      case (list: java.util.List[_], ARRAY) if typ.getArgumentsAsTypeSignatures.size() > 0 =>
```

Comment:
why the guard is needed here?

Reply:
> why the guard is needed here?

Have call `getArgumentsAsTypeSignatures .get(0)` below, used to avoid out of index. However, it may not be necessary, because `getArgumentsAsTypeSignatures.get(0)` always has a value.
Have removed it.

Reply:
so shall we `assert(getArgumentsAsTypeSignatures.nonEmpty)` first ?

Reply:
sure, done.